driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java

```diff
@@ -116,11 +117,14 @@ static WriteConcern asWriteConcern(final BsonDocument writeConcernDocument) {
     }
 
     public static ReadPreference asReadPreference(final BsonDocument readPreferenceDocument) {
-        if (readPreferenceDocument.size() > 1) {
-            throw new UnsupportedOperationException("Unsupported read preference properties");
+        if (readPreferenceDocument.size() == 1) {
```

Comment:
For some reason the new tests use `maxStalenessSeconds` so had to add support for it.